### PR TITLE
Introducing automation e2e tests

### PR DESCRIPTION
**Pull request by jalkire**
#### Describe the change
- Add an E2E test to AutomationTests
- [Crossing the bridge](https://github.com/microsoft/axe-windows/pull/48#issuecomment-499989796) to update the check-automation script to use the new scan interface

> Note: After the PR has been created, certain checks will be kicked off. All of these checks must pass before a merge.

### CODE REVIEW

File: src/AutomationTests/AutomationIntegrationTests.cs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+                .Build();
+
+            var scanner = ScannerFactory.CreateScanner(config);
+
+            var output = scanner.Scan();
+
+            var regexForExpectedFile = $"{OutputDir.Replace("\\", "\\\\")}.*\\.a11ytest";
+
+            // Validate that we got the expected error count
+            Assert.AreEqual(12, output.ErrorCount);
```
Comment (DaveTryon) [inline src/AutomationTests/AutomationIntegrationTests.cs]: >12 [](start = 28, length = 2)

Having a hardcoded value here has the potential to break whenever rules change.

File: src/AutomationTests/AutomationIntegrationTests.cs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            Assert.AreEqual(12, output.Errors.Count());
```
Comment (DaveTryon) [inline src/AutomationTests/AutomationIntegrationTests.cs]: Can we remove the duplicated constant?

```suggestion
            Assert.AreEqual(output.ErrorCount, output.Errors.Count());
```

File: src/AutomationTests/AutomationIntegrationTests.cs (hunk 1 above)
Comment (jalkire) [reply src/AutomationTests/AutomationIntegrationTests.cs]: Updated to just make sure that there are > 0 errors. If there are ever 0, we should probably update the test app.

File: src/AutomationTests/AutomationIntegrationTests.cs (hunk 2 above)
Comment (jalkire) [reply src/AutomationTests/AutomationIntegrationTests.cs]: Yes

**Pull request merged**